### ISSUE TITLE
junk data

### ISSUE THREAD

**Issue by Tcll** [ipfs/js-ipfs#3346]
after 2 failed uploads containing oversized files in some data I was backing up, I now have a repo that's almost twice the size it should be.

if I add up the size of all the data I've uploaded, the size comes around 17 to 18 GiB
yet when I check the properties of my blocks/ directory, the size is 31.1 GiB

is it possible to clean the junk data from the failed uploads to shrink my repo back to what it's supposed to be??

just to note, de-duplication absolutely works like it's supposed to on re-upload (no extra data is created)
but when you have files that are cancelled, the partially-uploaded data seems to remain there...
I would expect ipfs to clear the partial data when cancelled :/

EDIT:
btw, any suggestions involving creating a new repo won't due
imagine a circumstance involving a power outage during an upload
the repo will need post cleaning in much the same way as anything else that doesn't involve an ipfs restart
so in other words future fixes still need to deal with current junk data, if there isn't a way to deal with it already

I've done numerous searches on DDG and can't find anything dealing with my particular issue...

**Comment by achingbrain** [comment]
If you run gc on the repo it will remove any blocks that are not pinned - if your uploads were big the files will be chunked but only the root node will be pinned so you may have the intermediate blocks hanging around.

Please use https://discuss.ipfs.io/tag/js-ipfs for support questions in future, it's more searchable than GitHub which is really just for bugs and dev coordination.

**Comment by Tcll** [comment]
unfortunately the page doesn't load or I would
I don't really like using GitHub myself as it's not freedom focused

but I'll try that out... do I have to pin everything needed via the web UI just to ensure nothing important would be deleted??

**Comment by achingbrain** [comment]
> unfortunately the page doesn't load or I would

The link works for me, what do you see?

> do I have to pin everything needed via the web UI just to ensure nothing important would be deleted??

You can use the CLI too, there's more info here: https://docs.ipfs.io/how-to/pin-files/

> the web UI doesn't support pin selected or pin recursively...  that's annoying, especially when you have a folder containing 300 items

That does sound annoying - could you open an issue in https://github.com/ipfs-shipyard/ipfs-webui - they are working hard to improve the UX.

Are these files in your MFS?  That's pinned recursively by default.

**Comment by Tcll** [comment]
> what do you see?

sadly, just a blank white page that only loads about 20% and hangs indefinitely

> Are these files in your MFS? That's pinned recursively by default.

not sure, I usually manage things via the web UI (except when preserving mtime) just because it's easier than a terminal...
when preserving mtime, I pretty much just do an `add -r` and then import the key via the web UI.
(I really wish there was a C++/Qt or at least Py/Qt app for this)

> That does sound annoying - could you open an issue

sure thing :+1:
